sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -380,6 +380,9 @@ object ColumnPruning extends Rule[LogicalPlan] {
         p
       }
 
+    // Eliminate the Projects with empty projectList
+    case p @ Project(projectList, child) if projectList.isEmpty => child
```

Comment:
how about we just move that case ahead? It seems always safe to apply `case p @ Project(projectList, child) if sameOutput(child.output, p.output) => child`